tests/parallel_for/hierarchical.cpp

```diff
@@ -33,30 +33,31 @@ int main() {
   const int size = 10;
   std::vector<int> data(size);
   const int groupsize = 2;
-/* Put &data[0] instead of data.data() because it is not obvious in the
-   excerpt below it is a vector */
-//////// Start slide
-buffer<int> input(&data[0], size);
-buffer<int> output(size);
-my_queue.submit([&](handler &cgh)
-{
-  auto in_access = input.get_access<access::mode::read>(cgh);
-  auto out_access = output.get_access<access::mode::write>(cgh);
-
-  cgh.parallel_for_work_group<class hierarchical>(nd_range<>(range<>(size),
-                                                             range<>(groupsize)),
-                                                  [=](group<> group)
+  /* Put &data[0] instead of data.data() because it is not obvious in the
+     excerpt below it is a vector */
+  //////// Start slide
+  buffer<int> input(&data[0], size);
+  buffer<int> output(size);
+  my_queue.submit([&](handler &cgh)
   {
-    std::cout << "Group id = " << group.get(0) << std::endl;
+    auto in_access = input.get_access<access::mode::read>(cgh);
+    auto out_access = output.get_access<access::mode::write>(cgh);
 
-    parallel_for_work_item(group, [=](nd_item<1> tile)
+    cgh.parallel_for_work_group<class hierarchical>(nd_range<>(range<>(size),
+                                                               range<>(groupsize)),
+                                                    [=](group<> group)
     {
-      std::cout << "Local id = " << tile.get_local(0)
-                << " (global id = " << tile.get_global(0) << ")" << std::endl;
-      out_access[tile] = in_access[tile] * 2;
+      std::cout << "Group id = " << group.get(0) << std::endl;
+
+      parallel_for_work_item(group, [=](nd_item<1> tile)
+      {
+        std::cout << "Local id = " << tile.get_local(0)
+                  << " (global id = " << tile.get_global(0) << ")" << std::endl;
+        out_access[tile] = in_access[tile] * 2;
+      });
     });
   });
-});
-//////// End slide
-    return 0;
+  //////// End slide
+  my_queue.wait();
```

Comment:
There is no reason for output's destruction to be blocking as it does not trigger a write-back.